src/main/java/org/mifos/connector/ams/pesacore/camel/route/PesaRouteBuilder.java

```diff
@@ -62,7 +62,8 @@ public void configure() {
                 .setBody(e -> {
                     String body=e.getIn().getBody(String.class);
                     logger.debug("Body : {}",body);
-                    e.setProperty("dfspId",e.getProperty("dfspId"));
+                    String accountHoldingInstitutionId= String.valueOf(e.getIn().getHeader("accountHoldingInstitutionId"));
```

Comment:
why do we need accountHoldingInstitutionId in pesa?

Reply:
This gets passed from response back to mpesa->channel to start the workflow.